compiler/src/dotty/tools/dotc/util/Chars.scala

```diff
@@ -93,4 +93,13 @@ object Chars {
          '|' | '/' | '\\' => true
     case c => isSpecial(c)
   }
+
+  /** Would the character be encoded by `NameTransformer.encode`? */
+  def willBeEncoded(c : Char) : Boolean = (c: @switch) match {
+    case '~' | '!' | '@' | '#' | '%' |
+         '^' | '*' | '+' | '-' | '<' |
+         '>' | '?' | ':' | '=' | '&' |
+         '|' | '/' | '\\' => true
```

Comment:
`isJavaIdentifierPart` returns `false` for all those characters. So you can remove this.